### ISSUE TITLE
`rtree` causes crash during `Dataset` init

### ISSUE THREAD

**Issue by MaximeRM** [umr-amap/iamap#13]
Qgis crashes when testing the encoder. Error on the terminal below:

With python 3.12; Ubuntu 24.04.

Potentially related to this: https://github.com/qgis/QGIS/issues/56599

```
QGIS died on signal 11
This GDB supports auto-downloading debuginfo from the following URLs:
  <https://debuginfod.ubuntu.com>
Enable debuginfod for this session? (y or [n]) [answered N; input not from terminal]
Debuginfod has been disabled.
To make this setting permanent, add 'set debuginfod enabled off' to .gdbinit.

warning: could not find '.gnu_debugaltlink' file for /usr/bin/qgis.bin
Could not attach to process.  If your uid matches the uid of the target
process, check the setting of /proc/sys/kernel/yama/ptrace_scope, or try
again as the root user.  For more details, see /etc/sysctl.d/10-ptrace.conf
ptrace: Operation not permitted.
No thread selected
No stack.
gdb returned 256
Aborted (core dumped)
```

**Comment by ptresson** [comment]
I've reproduced the crash with a fresh conda environment with python 3.12. with defaut parameters during encoding. I'll see what can be used to work around this. The QGIS issue seems to suggest we are not the only ones having problems with python 3.12, it may be outside the scope of the plugin.

**Comment by ptresson** [comment]
crash happens during the creation of a torchgeo `RasterDatatset`more precisely. I'll try to see if it may be a question of memory allocation or a GIS related crash

**Comment by ptresson** [comment]
working env:
```
Python 3.11.9 | packaged by conda-forge | (main, Apr 19 2024, 18:36:13) [GCC 12.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import rasterio
>>> rasterio.show_versions()
rasterio info:
  rasterio: 1.3.9
      GDAL: 3.8.3
      PROJ: 9.3.1
      GEOS: 3.12.1
 PROJ DATA: /home/tresson/mambaforge/envs/geosam/share/proj
 GDAL DATA: /home/tresson/mambaforge/envs/geosam/share/gdal

System:
    python: 3.11.9 | packaged by conda-forge | (main, Apr 19 2024, 18:36:13) [GCC 12.3.0]
executable: /home/tresson/mambaforge/envs/geosam/bin/python
   machine: Linux-6.8.0-40-generic-x86_64-with-glibc2.35

Python deps:
    affine: 2.4.0
     attrs: 23.2.0
   certifi: 2024.02.02
     click: 8.1.7
     cligj: 0.7.2
    cython: None
     numpy: 1.26.4
    snuggs: 1.4.7
click-plugins: None
setuptools: 69.1.0
>>>
```

broken env:
```
import rasterio
rasterio.show_versions()
rasterio info:
  rasterio: 1.3.11
      GDAL: 3.9.2
      PROJ: 9.4.1
      GEOS: 3.12.2
 PROJ DATA: /home/tresson/mambaforge/envs/test/share/proj
 GDAL DATA: /home/tresson/mambaforge/envs/test/share/gdal

System:
    python: 3.12.5 | packaged by conda-forge | (main, Aug  8 2024, 18:57:17) [GCC 12.4.0]
executable: /home/tresson/mambaforge/envs/test/bin/python3
   machine: Linux-6.8.0-40-generic-x86_64-with-glibc2.35

Python deps:
    affine: 2.4.0
     attrs: 24.2.0
   certifi: 2024.08.30
     click: 8.1.7
     cligj: 0.7.2
    cython: None
     numpy: 2.1.1
    snuggs: 1.4.7
click-plugins: None
setuptools: 73.0.1
```

**Comment by ptresson** [comment]
related:
https://github.com/qgis/QGIS/issues/57320

creating an rtree index in the QGIS python console is sufficient to cause the crash

**Comment by ptresson** [comment]
a working solution seem to uninstall pip rtree and install it via package manager

https://geo-sam.readthedocs.io/en/latest/blog/2024/05-02_crash_on_QGIS.html
https://github.com/coolzhao/Geo-SAM/issues/43

**Comment by ptresson** [comment]
several solutions have been found:

*  limit to QGIS < 3.36 (which does seem a bad idea)
*  link to other ressources on how to solve the issue
*  For our usage, uninstalling `rtree > 1.0` *after* torchgeo install (since its a dependencie), then installing `rtree < 1.0` via pip seem to work. this will need post installation checkups but could work. Such post installation scripts could be of use for other conflicts such as proj and rasterio versions or others (see https://github.com/umr-amap/iamap/issues/17)

**Comment by ptresson** [comment]
seems to be fixed with 9500bf10c1c1506f746c1bc197ddb87f035a7947
the better long term solution would be not to rely on `rtree`.
I'll close this issue and open a new one with this.

**Comment by ptresson** [comment]
having `rtree <= 0.9.7` seem unstable in the long run. installing all dependencies via conda seems to work even for rtree > 1.
otherwise, it could be installed on the system, as suggested in related QGIS issue. Maybe an option would be to test after install if this crashes and redirect to help on how to solve the issue.

**Comment by ptresson** [comment]
Should be solved with c8147ec23182986ec2f0290094977f58a88662ce
